2020-06-11 17:51:07 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "CREATE TABLE NEWS(ID BIGINT PRIMARY KEY AUTO_INCREMENT,
TITLE TEXT,
CONTENT TEXT,
URL VARCHAR(100),
CREAT_AT TIMESTAMP,
MODIFIED_AT TIMESTAMP)
MODIFIED_AT[*] TIMESTAMP) "; SQL statement:
create table News(id bigint primary key auto_increment,
title text,
content text,
url varchar(100),
creat_at timestamp,
modified_at timestamp)
modified_at timestamp) [42000-199]
2020-06-11 17:54:20 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSEDS" not found; SQL statement:
select link from links_already_processedS [42102-199]
2020-06-11 17:56:32 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSEDS" not found; SQL statement:
select link from links_already_processedS [42102-199]
2020-06-12 13:44:05 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELETE TABLE[*] LINKS_ALREADY_PROCESSED "; expected "identifier"; SQL statement:
delete table LINKS_ALREADY_PROCESSED [42001-199]
2020-06-12 13:44:18 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELATE[*] TABLE LINKS_ALREADY_PROCESSED "; expected "DELETE, DROP, DECLARE, DEALLOCATE"; SQL statement:
delate table LINKS_ALREADY_PROCESSED [42001-199]
2020-06-12 13:44:37 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELETE TABLE[*] LINKS_ALREADY_PROCESSED "; expected "identifier"; SQL statement:
delete table LINKS_ALREADY_PROCESSED [42001-199]
2020-06-12 13:44:43 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELETE TABLE[*] FROM LINKS_ALREADY_PROCESSED "; expected "identifier"; SQL statement:
delete table from LINKS_ALREADY_PROCESSED [42001-199]
2020-06-12 13:45:55 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSED" already exists; SQL statement:
create table LINKS_ALREADY_PROCESSED (link varchar (1000)) [42101-199]
2020-06-12 13:46:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELATE[*] FROM  LINKS_TO_BE_PROCESSED WHERE LINK = ?  "; expected "DELETE, DROP, DECLARE, DEALLOCATE"; SQL statement:
DELATE FROM  LINKS_TO_BE_PROCESSED where link = ?  [42001-199]
2020-06-12 13:47:01 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_TO_BE_PROSESSED" not found; SQL statement:
INSERT INTO LINKS_TO_BE_PROSESSED(link) values (?) [42102-199]
2020-06-12 13:47:47 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_TO_BE_PROSESSED" not found; SQL statement:
INSERT INTO LINKS_TO_BE_PROSESSED(link) values (?) [42102-199]
2020-06-12 13:52:35 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:72)
2020-06-12 14:00:52 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:65)
2020-06-12 14:04:11 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:61)
2020-06-12 14:25:10 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientException: Method is only allowed for a query. Use execute or executeUpdate instead of executeQuery; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [90002-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.command.Prepared.query(Prepared.java:229)
	at org.h2.command.CommandContainer.query(CommandContainer.java:145)
	at org.h2.command.Command.executeQuery(Command.java:202)
	at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:115)
	at com.github.Hana.Main.main(Main.java:63)
2020-06-12 14:27:10 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(100)": "'https://cj.sina.cn/article/norm_detail?url=https%3A%2F%2Ffinance.sina.cn%2F2020-06-12%2Fdetail-iirczymk6592669.d.html&vt=4&pos=... (130)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 14:31:47 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:61)
2020-06-12 14:36:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:36:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:36:59 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:36:59 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:37:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:37:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:37:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:37:01 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:37:01 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:37:02 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSE" not found; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSE(link) values (?) [42102-199]
2020-06-12 14:38:49 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSEED" not found; SQL statement:
select * from LINKS_ALREADY_PROCESSEED [42102-199]
2020-06-12 14:39:24 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINK" not found; SQL statement:
delete link from LINKS_ALREADY_PROCESSED [42102-199]
2020-06-12 14:39:34 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS" not found; SQL statement:
delete links from LINKS_ALREADY_PROCESSED [42102-199]
2020-06-12 14:40:18 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:25)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-12 14:42:33 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "ELECT[*] * FROM LINKS_TO_BE_PROCESSED "; expected "EXPLAIN, EXECUTE"; SQL statement:
elect * from LINKS_TO_BE_PROCESSED [42001-199]
2020-06-12 14:51:39 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:25)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-12 15:25:10 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:25)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-12 15:32:32 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:25)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-12 16:01:48 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6537288.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:01:49 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iircuyvi8008044.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:01:50 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fwm%2F2020-06-12%2Fdetail-iircuyvi8007389.d.html%3Fvt%3D4%26pos%3D1... (2061)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:01:51 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fwm%2F2020-06-12%2Fdetail-iirczymk6536333.d.html%3Fvt%3D4%26pos%3D1... (2061)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:01:52 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6536334.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6534771.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:66)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:66)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:66)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:66)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:66)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:66)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.disposeUsedLinkFromDatabase(Main.java:95)
	at com.github.Hana.Main.main(Main.java:76)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:78)
2020-06-12 16:01:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.main(Main.java:49)
2020-06-12 16:05:25 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fpi%2Fpf%2F2014-11-24%2Fdetail-icesifvw8291824.d.html%3Fvt%3D4&titl... (2054)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:26:50 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6537288.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:26:51 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iircuyvi8008044.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:26:51 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fwm%2F2020-06-12%2Fdetail-iircuyvi8007389.d.html%3Fvt%3D4%26pos%3D1... (2061)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:26:52 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fwm%2F2020-06-12%2Fdetail-iirczymk6536333.d.html%3Fvt%3D4%26pos%3D1... (2061)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:26:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6536334.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:26:54 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6534771.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-12 16:27:31 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000) SELECTIVITY 20": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fpi%2Fpf%2F2014-11-24%2Fdetail-icesifvw8291824.d.html%3Fvt%3D4&titl... (2054)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 00:29:47 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000) SELECTIVITY 20": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6537288.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 00:32:41 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.insertLinkIntoDatabase(Main.java:98)
	at com.github.Hana.Main.parseUrlsFromPageAndStoreIntoDatabase(Main.java:77)
	at com.github.Hana.Main.main(Main.java:58)
2020-06-13 00:38:29 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000) SELECTIVITY 20": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6537288.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 16:24:48 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000) SELECTIVITY 20": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6537288.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 16:27:17 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1000) SELECTIVITY 20": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6537288.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 16:37:08 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at net.sourceforge.squirrel_sql.fw.sql.SQLDriverManager.getConnection(SQLDriverManager.java:147)
	at net.sourceforge.squirrel_sql.client.mainframe.action.OpenConnectionCommand.executeConnect(OpenConnectionCommand.java:148)
	at net.sourceforge.squirrel_sql.client.mainframe.action.OpenConnectionCommand.lambda$execute$0(OpenConnectionCommand.java:93)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 30 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 24 more
2020-06-13 16:55:41 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-13 17:01:22 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "http://www.12377.cn/" not found; SQL statement:
DELETE FROM  LINKS_TO_BE_PROCESSED where link = "http://www.12377.cn/" [42122-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:451)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.expression.ExpressionColumn.getColumnException(ExpressionColumn.java:176)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:158)
	at org.h2.expression.condition.Comparison.optimize(Comparison.java:207)
	at org.h2.command.dml.Delete.prepare(Delete.java:160)
	at org.h2.command.Parser.prepareCommand(Parser.java:689)
	at org.h2.engine.Session.prepareLocal(Session.java:627)
	at org.h2.engine.Session.prepareCommand(Session.java:565)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1292)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:205)
	at net.sourceforge.squirrel_sql.client.session.StatementWrapper.execute(StatementWrapper.java:168)
	at net.sourceforge.squirrel_sql.client.session.SQLExecuterTask.processQuery(SQLExecuterTask.java:355)
	at net.sourceforge.squirrel_sql.client.session.SQLExecuterTask.run(SQLExecuterTask.java:209)
	at net.sourceforge.squirrel_sql.fw.util.TaskExecuter.run(TaskExecuter.java:82)
	at java.lang.Thread.run(Thread.java:748)
2020-06-13 17:02:33 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "http://www.12377.cn/" not found; SQL statement:
DELETE FROM  LINKS_TO_BE_PROCESSED where link = "http://www.12377.cn/" [42122-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:451)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.expression.ExpressionColumn.getColumnException(ExpressionColumn.java:176)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:158)
	at org.h2.expression.condition.Comparison.optimize(Comparison.java:207)
	at org.h2.command.dml.Delete.prepare(Delete.java:160)
	at org.h2.command.Parser.prepareCommand(Parser.java:689)
	at org.h2.engine.Session.prepareLocal(Session.java:627)
	at org.h2.engine.Session.prepareCommand(Session.java:565)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1292)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:205)
	at net.sourceforge.squirrel_sql.client.session.StatementWrapper.execute(StatementWrapper.java:168)
	at net.sourceforge.squirrel_sql.client.session.SQLExecuterTask.processQuery(SQLExecuterTask.java:355)
	at net.sourceforge.squirrel_sql.client.session.SQLExecuterTask.run(SQLExecuterTask.java:209)
	at net.sourceforge.squirrel_sql.fw.util.TaskExecuter.run(TaskExecuter.java:82)
	at java.lang.Thread.run(Thread.java:748)
2020-06-13 17:03:35 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "https://sina.cn/?vt=4&pos=108" not found; SQL statement:
DELETE FROM  LINKS_TO_BE_PROCESSED where link = "https://sina.cn/?vt=4&pos=108" [42122-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:451)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.expression.ExpressionColumn.getColumnException(ExpressionColumn.java:176)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:158)
	at org.h2.expression.condition.Comparison.optimize(Comparison.java:207)
	at org.h2.command.dml.Delete.prepare(Delete.java:160)
	at org.h2.command.Parser.prepareCommand(Parser.java:689)
	at org.h2.engine.Session.prepareLocal(Session.java:627)
	at org.h2.engine.Session.prepareCommand(Session.java:565)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1292)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:205)
	at net.sourceforge.squirrel_sql.client.session.StatementWrapper.execute(StatementWrapper.java:168)
	at net.sourceforge.squirrel_sql.client.session.SQLExecuterTask.processQuery(SQLExecuterTask.java:355)
	at net.sourceforge.squirrel_sql.client.session.SQLExecuterTask.run(SQLExecuterTask.java:209)
	at net.sourceforge.squirrel_sql.fw.util.TaskExecuter.run(TaskExecuter.java:82)
	at java.lang.Thread.run(Thread.java:748)
2020-06-13 17:31:29 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-13 17:31:47 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-13 17:54:39 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.disposeUsedLinkFromDatabase(Main.java:109)
	at com.github.Hana.Main.main(Main.java:46)
2020-06-13 18:25:45 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-13 19:04:32 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-13 19:05:55 database: could not close the database
java.lang.IllegalStateException: Transaction was illegally transitioned from COMMITTED to ROLLED_BACK [1.4.199/103]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.tx.Transaction.setStatus(Transaction.java:222)
	at org.h2.mvstore.tx.Transaction.rollback(Transaction.java:444)
	at org.h2.engine.Session.rollbackTo(Session.java:827)
	at org.h2.engine.Session.rollback(Session.java:797)
	at org.h2.engine.Session.close(Session.java:891)
	at org.h2.engine.Database.closeAllSessionsException(Database.java:1400)
	at org.h2.engine.Database.closeImpl(Database.java:1454)
	at org.h2.engine.Database.close(Database.java:1417)
	at org.h2.engine.OnExitDatabaseCloser.onShutdown(OnExitDatabaseCloser.java:85)
	at org.h2.engine.OnExitDatabaseCloser.run(OnExitDatabaseCloser.java:114)
	Suppressed: java.lang.IllegalStateException: Transaction was illegally transitioned from CLOSED to CLOSED [1.4.199/103]
		at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
		at org.h2.mvstore.tx.Transaction.setStatus(Transaction.java:222)
		at org.h2.mvstore.tx.Transaction.closeIt(Transaction.java:507)
		at org.h2.mvstore.tx.TransactionStore.endTransaction(TransactionStore.java:549)
		at org.h2.mvstore.tx.Transaction.rollback(Transaction.java:454)
		... 8 more
2020-06-13 19:10:23 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Connection is broken: "session closed" [90067-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.engine.Session.prepareLocal(Session.java:604)
	at org.h2.engine.Session.prepareCommand(Session.java:565)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1292)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:77)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:349)
	at com.github.Hana.Main.insertLinkIntoDatabase(Main.java:100)
	at com.github.Hana.Main.main(Main.java:44)
2020-06-13 19:14:07 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELETE FROM LINKS_TO_BE_PROCESSED([*]LINK) VALUES (?) "; SQL statement:
DELETE FROM LINKS_TO_BE_PROCESSED(link) values (?) [42000-199]
2020-06-13 19:15:16 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELETE FROM LINKS_TO_BE_PROCESSED([*]LINK) VALUES (?) "; SQL statement:
DELETE FROM LINKS_TO_BE_PROCESSED(link) values (?) [42000-199]
2020-06-13 19:24:11 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELETEFROM[*] LINKS_TO_BE_PROCESSED(LINK) VALUES (?) "; expected "DELETE, DROP, DECLARE, DEALLOCATE"; SQL statement:
DELETEFROM LINKS_TO_BE_PROCESSED(link) values (?) [42001-199]
2020-06-13 19:24:26 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "DELETE FROM LINKS_TO_BE_PROCESSED([*]LINK) VALUES (?) "; SQL statement:
DELETE FROM LINKS_TO_BE_PROCESSED(link) values (?) [42000-199]
2020-06-13 19:26:33 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcStatement.checkClosed(JdbcStatement.java:1237)
	at org.h2.jdbc.JdbcPreparedStatement.checkClosed(JdbcPreparedStatement.java:1829)
	at org.h2.jdbc.JdbcStatement.checkClosed(JdbcStatement.java:1211)
	at org.h2.jdbc.JdbcResultSet.checkClosed(JdbcResultSet.java:3245)
	at org.h2.jdbc.JdbcResultSet.checkColumnIndex(JdbcResultSet.java:3229)
	at org.h2.jdbc.JdbcResultSet.get(JdbcResultSet.java:3270)
	at org.h2.jdbc.JdbcResultSet.getString(JdbcResultSet.java:296)
	at com.github.Hana.Main.loadUrlFromDatabase(Main.java:118)
	at com.github.Hana.Main.main(Main.java:35)
2020-06-13 19:37:45 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1100)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6537288.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 19:40:47 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1100)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iircuyvi8008044.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 19:40:57 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1100)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fwm%2F2020-06-12%2Fdetail-iircuyvi8007389.d.html%3Fvt%3D4%26pos%3D1... (2061)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 19:41:31 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1100)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fwm%2F2020-06-12%2Fdetail-iirczymk6536333.d.html%3Fvt%3D4%26pos%3D1... (2061)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 19:42:22 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(1100)": "'http://share.sina.cn/callback?vt=4&url=%2F%2Fgames.sina.cn%2Fgn%2Fvi%2F2020-06-12%2Fdetail-iirczymk6534771.d.html%3Fvt%3D4%26po... (2066)"; SQL statement:
INSERT INTO LINKS_TO_BE_PROCESSED(link) values (?) [22001-199]
2020-06-13 19:45:48 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement ")[*])"; SQL statement:
)) [42000-199]
2020-06-13 19:45:54 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_TO_BE_PROCESSED" already exists; SQL statement:
create table LINKS_TO_BE_PROCESSED (link varchar (2100)) [42101-199]
2020-06-13 19:55:28 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-13 19:59:27 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-13 21:55:28 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: File corrupted while reading record: "nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db". Possible solution: use the recovery tool; SQL statement:
DELETE FROM LINKS_TO_BE_PROCESSED where link = ? [90030-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.mvstore.db.MVTableEngine$Store.convertIllegalStateException(MVTableEngine.java:205)
	at org.h2.mvstore.db.MVTable.convertException(MVTable.java:725)
	at org.h2.mvstore.db.MVPrimaryIndex.lockRow(MVPrimaryIndex.java:235)
	at org.h2.mvstore.db.MVPrimaryIndex.lockRow(MVPrimaryIndex.java:227)
	at org.h2.mvstore.db.MVTable.lockRow(MVTable.java:582)
	at org.h2.command.dml.Delete.update(Delete.java:95)
	at org.h2.command.CommandContainer.update(CommandContainer.java:133)
	at org.h2.command.Command.executeUpdate(Command.java:267)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:200)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:154)
	at com.github.Hana.Main.deleteLinkFromDatabase(Main.java:102)
	at com.github.Hana.Main.main(Main.java:40)
Caused by: java.lang.IllegalStateException: This store is closed [1.4.199/4]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.MVStore.checkOpen(MVStore.java:2634)
	at org.h2.mvstore.MVStore.getMapName(MVStore.java:2736)
	at org.h2.mvstore.MVMap.beforeWrite(MVMap.java:889)
	at org.h2.mvstore.MVMap.append(MVMap.java:1310)
	at org.h2.mvstore.tx.TransactionStore.addUndoLogRecord(TransactionStore.java:413)
	at org.h2.mvstore.tx.Transaction.log(Transaction.java:306)
	at org.h2.mvstore.tx.TxDecisionMaker.logIt(TxDecisionMaker.java:131)
	at org.h2.mvstore.tx.TxDecisionMaker.decide(TxDecisionMaker.java:69)
	at org.h2.mvstore.tx.TxDecisionMaker$LockDecisionMaker.decide(TxDecisionMaker.java:286)
	at org.h2.mvstore.tx.TxDecisionMaker$LockDecisionMaker.decide(TxDecisionMaker.java:278)
	at org.h2.mvstore.MVMap.operate(MVMap.java:1670)
	at org.h2.mvstore.tx.TransactionMap.set(TransactionMap.java:306)
	at org.h2.mvstore.tx.TransactionMap.lock(TransactionMap.java:268)
	at org.h2.mvstore.db.MVPrimaryIndex.lockRow(MVPrimaryIndex.java:233)
	... 9 more
2020-06-14 12:41:16 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """LINK"" VARCHAR(100)": "'https://cj.sina.cn/article/normal_detail?url=https%3A%2F%2Ffinance.sina.cn%2F2020-06-12%2Fdetail-iircuyvi8163522.d.html&vt=4&po... (132)"; SQL statement:
INSERT INTO LINKS_ALREADY_PROCESSED(link) values (?) [22001-199]
2020-06-14 14:33:27 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "LINKS_ALREADY_PROCESSED" not found; SQL statement:
select * from LINKS_ALREADY_PROCESSED [42102-199]
2020-06-14 14:49:17 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:30)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-14 14:49:56 jdbc[3]: exception
org.h2.jdbc.JdbcSQLNonTransientConnectionException: Database is already closed (to disable automatic closing at VM shutdown, add ";DB_CLOSE_ON_EXIT=FALSE" to the db URL) [90121-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:617)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.message.DbException.get(DbException.java:170)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1571)
	at org.h2.jdbc.JdbcConnection.checkClosed(JdbcConnection.java:1547)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:347)
	at com.github.Hana.Main.updateDatabate(Main.java:77)
	at com.github.Hana.Main.parseUrlsFromPageAndStoreIntoDatabase(Main.java:55)
	at com.github.Hana.Main.main(Main.java:43)
2020-06-14 15:54:10 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "COUNT[*](*) FROM LINKS_TO_BE_PROCESSED "; expected "COMMIT, CREATE, CALL, CHECKPOINT, COMMENT"; SQL statement:
count(*) from links_to_be_processed [42001-199]
2020-06-14 15:54:19 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "COUNT[*][*] FROM LINKS_TO_BE_PROCESSED "; expected "COMMIT, CREATE, CALL, CHECKPOINT, COMMENT"; SQL statement:
count[*] from links_to_be_processed [42001-199]
2020-06-14 15:54:24 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "COUNT[*] FROM LINKS_TO_BE_PROCESSED "; expected "COMMIT, CREATE, CALL, CHECKPOINT, COMMENT"; SQL statement:
count from links_to_be_processed [42001-199]
2020-06-14 15:55:17 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "SESCT[*] COUNT(*) FROM LINKS_TO_BE_PROCESSED "; expected "SET, SAVEPOINT, SCRIPT, SHUTDOWN, SHOW"; SQL statement:
sesct count(*) from links_to_be_processed [42001-199]
2020-06-14 15:55:19 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Syntax error in SQL statement "SESCT[*] COUNT(*) FROM LINKS_TO_BE_PROCESSED "; expected "SET, SAVEPOINT, SCRIPT, SHUTDOWN, SHOW"; SQL statement:
sesct count(*) from links_to_be_processed [42001-199]
2020-06-14 16:08:07 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:433)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnection(DriverDataSource.java:399)
	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:56)
	at org.flywaydb.core.internal.database.DatabaseFactory.createDatabase(DatabaseFactory.java:72)
	at org.flywaydb.core.Flyway.execute(Flyway.java:1670)
	at org.flywaydb.core.Flyway.clean(Flyway.java:1499)
	at org.flywaydb.maven.CleanMojo.doExecute(CleanMojo.java:35)
	at org.flywaydb.maven.AbstractFlywayMojo.execute(AbstractFlywayMojo.java:746)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 52 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 46 more
2020-06-14 16:08:18 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnectionFromDriver(DriverDataSource.java:433)
	at org.flywaydb.core.internal.jdbc.DriverDataSource.getConnection(DriverDataSource.java:399)
	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:56)
	at org.flywaydb.core.internal.database.DatabaseFactory.createDatabase(DatabaseFactory.java:72)
	at org.flywaydb.core.Flyway.execute(Flyway.java:1670)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:1356)
	at org.flywaydb.maven.MigrateMojo.doExecute(MigrateMojo.java:35)
	at org.flywaydb.maven.AbstractFlywayMojo.execute(AbstractFlywayMojo.java:746)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 52 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 46 more
2020-06-14 16:24:05 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.message.DbException.convert(DbException.java:347)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:90)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2787)
	at org.h2.mvstore.MVStore.panic(MVStore.java:441)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:404)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:162)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:95)
	at org.h2.engine.Database.getPageStore(Database.java:2739)
	at org.h2.engine.Database.open(Database.java:769)
	at org.h2.engine.Database.openDatabase(Database.java:319)
	at org.h2.engine.Database.<init>(Database.java:313)
	at org.h2.engine.Engine.openSession(Engine.java:69)
	at org.h2.engine.Engine.openSession(Engine.java:201)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:178)
	at org.h2.engine.Engine.createSession(Engine.java:161)
	at org.h2.engine.Engine.createSession(Engine.java:31)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:336)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:169)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:148)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at com.github.Hana.Main.main(Main.java:31)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: General error: "java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]" [50000-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 25 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:/Users/jinhua/IdeaProjects/news-crewler/news.mv.db [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883)
	at org.h2.mvstore.FileStore.open(FileStore.java:172)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390)
	... 19 more
2020-06-14 16:24:45 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "CREATED_AT" not found; SQL statement:
insert into news(url,title,content,created_at,modified_at) values(?,?,?,now(),now()) [42122-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:451)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.table.Table.getColumn(Table.java:708)
	at org.h2.command.Parser.parseColumn(Parser.java:1377)
	at org.h2.command.Parser.parseColumnList(Parser.java:1362)
	at org.h2.command.Parser.parseInsertGivenTable(Parser.java:1731)
	at org.h2.command.Parser.parseInsert(Parser.java:1684)
	at org.h2.command.Parser.parsePrepared(Parser.java:891)
	at org.h2.command.Parser.parse(Parser.java:788)
	at org.h2.command.Parser.parse(Parser.java:760)
	at org.h2.command.Parser.prepareCommand(Parser.java:683)
	at org.h2.engine.Session.prepareLocal(Session.java:627)
	at org.h2.engine.Session.prepareCommand(Session.java:565)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1292)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:77)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:349)
	at com.github.Hana.Main.storeIntoDatabaseIfItIsNewsPage(Main.java:127)
	at com.github.Hana.Main.main(Main.java:46)
2020-06-14 16:25:59 jdbc[3]: exception
org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "CREATE_AT" not found; SQL statement:
insert into news(url,title,content,create_at,modified_at) values(?,?,?,now(),now()) [42122-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:451)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.table.Table.getColumn(Table.java:708)
	at org.h2.command.Parser.parseColumn(Parser.java:1377)
	at org.h2.command.Parser.parseColumnList(Parser.java:1362)
	at org.h2.command.Parser.parseInsertGivenTable(Parser.java:1731)
	at org.h2.command.Parser.parseInsert(Parser.java:1684)
	at org.h2.command.Parser.parsePrepared(Parser.java:891)
	at org.h2.command.Parser.parse(Parser.java:788)
	at org.h2.command.Parser.parse(Parser.java:760)
	at org.h2.command.Parser.prepareCommand(Parser.java:683)
	at org.h2.engine.Session.prepareLocal(Session.java:627)
	at org.h2.engine.Session.prepareCommand(Session.java:565)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1292)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:77)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:349)
	at com.github.Hana.Main.storeIntoDatabaseIfItIsNewsPage(Main.java:127)
	at com.github.Hana.Main.main(Main.java:46)
2020-06-14 16:27:31 jdbc[3]: exception
org.h2.jdbc.JdbcSQLDataException: Value too long for column """URL"" VARCHAR(100)": "'https://cj.sina.cn/article/normal_detail?url=https%3A%2F%2Ffinance.sina.cn%2Fchina%2Fgncj%2F2020-06-12%2Fdetail-iirczymk6696507... (147)"; SQL statement:
insert into news(url,title,content,created_at,modified_at) values(?,?,?,now(),now()) [22001-199]
